File: web/myLibraries/util/geometry/DCEL/Face.js
"use strict"

/*
 * Face.java
 *
 * Version:
 *     $1.0$
 *
 * Revisions:
 *     $1.0$
 */

/**
 * Data structure of Face of DCEL
 *
 * The source code in java is from my own github:
 * @see <a href=https://github.com/fengkeyleaf/Algorithm/blob/main/ComputationalGeometry/ApplicationOfTriangulation/myLibraries/util/geometry/DCEL/Face.java>Face</a>
 *
 * @author       Xiaoyu Tongyang, or call me sora for short
 */

export default class Face {
    static IDStatic = 0;

    /**
     * constructs to create an instance of Vertex
     *
     * @param {HalfEdge} outComponent
     * */

    constructor( outComponent ) {
        this.outComponent = outComponent;
        this.innerComponents = [];
        this.ID = Face.IDStatic++;
        this.IDOfDualVertex = -1;
    }

    static resetIDStatic() {
        Face.IDStatic = 0;
    }

    /**
     * @param {HalfEdge} halfEdge
     * */

    addInnerComponent( halfEdge ) {
        this.innerComponents.add( halfEdge );
    }
}

File: Java/myLibraries/util/geometry/DCEL/Face.java
package myLibraries.util.geometry.DCEL;

/*
 * Face.java
 *
 * Version:
 *     $1.0$
 *
 * Revisions:
 *     $1.0$
 */

import java.util.ArrayList;
import java.util.List;

/**
 * Data structure of Face of DCEL
 *
 * @author       Xiaoyu Tongyang, or call me sora for short
 */

public class Face {
    private static int IDStatic = 0;
    public final int ID;
    public HalfEdge outComponent;
    public final List<HalfEdge> innerComponents = new ArrayList<>();
    // half-edge that the infinite face only has.
    private HalfEdge infiniteBoundary;

    public int IDOfDualVertex;

    /**
     * constructs to create an instance of Vertex
     * */

    public Face() { ID = IDStatic++; }

    public Face( HalfEdge outComponent ) {
        this.outComponent = outComponent;
        ID = IDStatic++;
    }

    public static
    void resetIDStatic() {
        IDStatic = 0;
    }

    public void addInnerComponent( HalfEdge halfEdge ) {
        innerComponents.add( halfEdge );
    }
}
